loadgen/helpers.go

```diff
@@ -322,29 +400,73 @@ func (d normalDistribution[T]) GetType() string {
 }
 
 func (d normalDistribution[T]) Validate() error {
-	if int64(d.stdDev) <= 0 {
-		return fmt.Errorf("standard deviation must be positive, got %v", d.stdDev)
-	}
-
-	if int64(d.max) < int64(d.min) {
-		return fmt.Errorf("max value (%v) cannot be less than min value (%v)", d.max, d.min)
+	switch any(d.stdDev).(type) {
+	case int64:
+		if any(d.stdDev).(int64) <= 0 {
+			return fmt.Errorf("standard deviation must be positive, got %v", d.stdDev)
+		}
+		if any(d.max).(int64) < any(d.min).(int64) {
+			return fmt.Errorf("max value (%v) cannot be less than min value (%v)", d.max, d.min)
+		}
+	case float32:
+		if any(d.stdDev).(float32) <= 0 {
+			return fmt.Errorf("standard deviation must be positive, got %v", d.stdDev)
+		}
+		if any(d.max).(float32) < any(d.min).(float32) {
+			return fmt.Errorf("max value (%v) cannot be less than min value (%v)", d.max, d.min)
+		}
+	case time.Duration:
+		if any(d.stdDev).(time.Duration) <= 0 {
+			return fmt.Errorf("standard deviation must be positive, got %v", d.stdDev)
+		}
+		if any(d.max).(time.Duration) < any(d.min).(time.Duration) {
+			return fmt.Errorf("max value (%v) cannot be less than min value (%v)", d.max, d.min)
+		}
 	}
-
 	return nil
 }
 
 func (d normalDistribution[T]) Sample() (T, bool) {
```

Comment:
This function is the most disturbingly "Go" code I've seen in a while

Reply:
😬